integration_tests/test_ibc.py

```diff
@@ -1,182 +1,156 @@
 import base64
 import json
-import subprocess
-from pathlib import Path
-from typing import NamedTuple
 
 import pytest
-from pystarport import ports
 
-from .network import Chainmain, Cronos, Hermes, setup_custom_cronos
+from .ibc_utils import RATIO, assert_ready, get_balance, prepare, prepare_network
 from .utils import (
     ADDRS,
     CONTRACTS,
     deploy_contract,
     eth_to_bech32,
     send_transaction,
-    supervisorctl,
     wait_for_fn,
     wait_for_new_blocks,
-    wait_for_port,
 )
 
 
-class IBCNetwork(NamedTuple):
-    cronos: Cronos
-    chainmain: Chainmain
-    hermes: Hermes
-
-
 @pytest.fixture(scope="module")
 def ibc(request, tmp_path_factory):
     "start-cronos"
-    path = tmp_path_factory.mktemp("ibc")
-    gen = setup_custom_cronos(
-        path, 26700, Path(__file__).parent / "configs/ibc.jsonnet"
-    )
-    cronos = next(gen)
+    name = "ibc"
+    path = tmp_path_factory.mktemp(name)
+    network = prepare_network(path, name)
     try:
-        chainmain = Chainmain(cronos.base_dir.parent / "chainmain-1")
-        hermes = Hermes(cronos.base_dir.parent / "relayer.toml")
-        # wait for grpc ready
-        wait_for_port(ports.grpc_port(chainmain.base_port(0)))  # chainmain grpc
-        wait_for_port(ports.grpc_port(cronos.base_port(0)))  # cronos grpc
-        subprocess.check_call(
-            [
-                "hermes",
-                "-c",
-                hermes.configpath,
-                "create",
-                "channel",
-                "cronos_777-1",
-                "chainmain-1",
-                "--port-a",
-                "transfer",
-                "--port-b",
-                "transfer",
-            ]
-        )
-        supervisorctl(cronos.base_dir / "../tasks.ini", "start", "relayer-demo")
-        wait_for_port(hermes.port)
-        yield IBCNetwork(cronos, chainmain, hermes)
+        yield next(network)
     finally:
         try:
-            next(gen)
+            next(network)
         except StopIteration:
             pass
 
 
-def get_balance(chain, addr, denom):
-    return chain.cosmos_cli().balance(addr, denom)
-
-
 def get_balances(chain, addr):
```

Comment:
you can also move this one to ibc_utils

Reply:
only one place use, maybe mv to reuse when needed